actions/apps.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Client4} from 'mattermost-redux/client';
+import {submitInteractiveDialog as submitInteractiveDialogRedux} from 'mattermost-redux/actions/integrations';
+import {DialogSubmission} from 'mattermost-redux/types/integrations';
+import {getCurrentChannelId} from 'mattermost-redux/selectors/entities/common';
+import {ActionFunc, DispatchFunc, GetStateFunc} from 'mattermost-redux/types/actions';
+import {CallResponseTypes, CallResponse, Call} from 'mattermost-redux/types/apps';
+
+import {sendEphemeralPost} from 'actions/global_actions';
+import {openInteractiveDialog} from 'plugins/interactive_dialog';
+import {executeCommand} from 'actions/command';
+
+export function doAppCall(call: Call): ActionFunc {
+    return async (dispatch: DispatchFunc) => {
+        const res = await Client4.executeAppCall(call) as CallResponse;
+
+        if (res.markdown) {
+            if (!call.context.channel_id) {
+                return {data: false};
+            }
+
+            sendEphemeralPost(res.markdown, call.context.channel_id, call.context.root_id);
+            return {data: true};
+        }
+
+        switch (res.type) {
+        case CallResponseTypes.MODAL:
+            if (!res.data) {
+                return {data: false};
+            }
+            res.data.app_id = call.context.app_id;
+            openInteractiveDialog(res.data);
+            return {data: res};
+        case CallResponseTypes.COMMAND:
```

Comment:
Note to self - I think I like it, is this a new capability?

Reply:
The theory was already in place (am app can run a command just by passing the call) but extended to any command.

Reply:
yy, we should treat both consistently from scoping perspective. Command would (in theory) give access to other plugin and built-in commands, no?

Reply:
Exact. It would execute any command available to that user.